### PR TITLE
Fix shbang parsing edge cases.

### LINKED ISSUE #11: yath won't run if perl is in #! twice
```
$ cat ~/tmp/test.t
#!/perl/blah/perl

use Test::More;

diag "Perl is $^X";
pass;

done_testing;
```

```
$ yath ~/tmp/test.t
[LAUNCH]  _  /Users/schwern/tmp/test.t
[STDERR]  +  Can't open perl script "/blah/perl": No such file or directory
[ PLAN ]  =  No plan was ever set.
[FAILED]  =  /Users/schwern/tmp/test.t

=== FAILURE SUMMARY ===
 * /Users/schwern/tmp/test.t
```

I noticed this while trying yath with Regexp::Common. It has `#!/opt/perl/bin/perl`. I'm wondering why yath is even trying to open the #! perl.

### PR DESCRIPTION
**Pull request by schwern**
This borrows from TAP::Parser::SourceHandler::Perl

Fixes #11

### CODE REVIEW

File: lib/Test2/Harness/Runner.pm [hunk 1]
```diff
@@ -124,19 +147,16 @@ sub header {
     );
 
     open(my $fh, '<', $file) or die "Could not open file $file: $!";
-    my $ln = 0;
-    while (my $line = <$fh>) {
-        $ln++;
+
+    my $shbang = $self->_parse_shbang(<$fh>);
```
Comment (exodist) [inline lib/Test2/Harness/Runner.pm]: This assumes the first line of the test is a #!, what if the test has no #! line and instead starts with a directive tot he harness? This will miss it.

Comment (exodist) [inline lib/Test2/Harness/Runner.pm]: This is something I thought about, and intended to test for, but never got around to it.

Comment (schwern) [inline lib/Test2/Harness/Runner.pm]: Good point, I'll fix that.

**Comment by schwern** [comment]
Fixed!

**Comment by exodist** [comment]
+1
